craft_parts/infos.py

```diff
@@ -269,6 +270,24 @@ def partitions(self) -> list[str] | None:
         """Return the project's partitions."""
         return self._partitions
 
+    @property
+    def default_partition(self) -> str | None:
+        """Get the "default" partition from a partition list."""
+        if self._partitions:
+            return self._partitions[0]
+        return None
+
+    def is_default_partition(self, partition: str | None) -> bool:
+        """Check if given partition is the default one."""
+        if self._partitions is None and (
+            partition is None or partition == DEFAULT_PARTITION
+        ):
+            return True
+        if self._partitions is not None:
+            return partition == self._partitions[0]
```

Comment:
When we have a real partition set as the default, using `(default)` should also work (even if not explicitly declared in the partition list -- using `default` and the default parition name should be equivalent). The case is `self._partitions` is not `None`, and `partition == DEFAULT_PARTITION`.

In practice this would only affect `organize`, because internally the default partition name is normalized to the first item in the partition list. Only when organizing the user would refer to the default partition by its name, or literal `(default)`, or nothing. But since this is a method to say if a given partition is the default one, it seems reasonable that it would accept the literal `default` name.